sources/packages/backend/apps/api/src/route-controllers/application/application.institutions.controller.ts

```diff
@@ -0,0 +1,59 @@
+import { Controller, Get, Param, ParseIntPipe } from "@nestjs/common";
+import { ApplicationService } from "../../services";
+import BaseController from "../BaseController";
+import { ApplicationBaseAPIOutDTO } from "./models/application.dto";
+import {
+  AllowAuthorizedParty,
+  HasStudentDataAccess,
+  IsBCPublicInstitution,
+  UserToken,
+} from "../../auth/decorators";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { ApiTags } from "@nestjs/swagger";
+import { ClientTypeBaseRoute } from "../../types";
+import { ApplicationControllerService } from "./application.controller.service";
+import { IInstitutionUserToken } from "../../auth";
+
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@IsBCPublicInstitution()
+@Controller("application")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-application`)
+export class ApplicationInstitutionsController extends BaseController {
+  constructor(
+    private readonly applicationService: ApplicationService,
+    private readonly applicationControllerService: ApplicationControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * API to fetch application details by applicationId.
+   * This API will be used by ministry users.
+   * @param applicationId
+   * @param studentId
+   * @returns Application details.
+   */
+  @HasStudentDataAccess("studentId")
```

Comment:
Trying to follow what we did for the locations, I believe that the student should be at the root of the endpoint, for instance, similar to what we did for the below one.
https://github.com/bcgov/SIMS/blob/main/sources/packages/backend/apps/api/src/route-controllers/program-info-request/program-info-request.institutions.controller.ts#L84

To get more context, search the source code by "@HasLocationAccess("locationId")" and check how we did it.

For this case, for instance, it would be

```
api/institutions/institution/student/:studentId/application/:application
```

@ann-aot @andrepestana-aot @sh16011993 @dheepak-aot thoughts?

Reply:
I agree on the basis that it puts the context more clearly and I believe in that case, this end point must move to student.institutions.controller right?

Reply:
did you mean `api/institutions/student/:studentId/application/:application` ?, if it s `api/institutions/institution/student/:studentId/application/:application` it will be `institution.institutions.controller` right?

Reply:
@ann-aot @dheepak-aot trying to add more information.
Currently, we have already implemented the below endpoints:
- api/institutions/student/251
- api/institutions/student/251/application-summary
- api/institutions/student/251/documents
- api/institutions/overaward/student/251/balance (this one is already not following the others)

The suggested one would be as below (as Ann mentioned)
- api/institutions/student/251/application/:application

Reply:
As discussed, changed the controller